auth/app/src/main/java/com/google/firebase/quickstart/auth/KotlinMainActivity.kt

```diff
@@ -0,0 +1,230 @@
+
+                // Name, email address, and profile photo Url
+                val name = profile.displayName
+                val email = profile.email
+                val photoUrl = profile.photoUrl
+            }
+        }
+        // [END get_provider_data]
+    }
+
+    override fun updateProfile() {
+        // [START update_profile]
+        val user = FirebaseAuth.getInstance().currentUser
+
+        val profileUpdates = UserProfileChangeRequest.Builder()
+                .setDisplayName("Jane Q. User")
+                .setPhotoUri(Uri.parse("https://example.com/jane-q-user/profile.jpg"))
+                .build()
+
+        user?.updateProfile(profileUpdates)
+                ?.addOnCompleteListener { task ->
+                    if (task.isSuccessful) {
+                        Log.d(TAG, "User profile updated.")
+                    }
+                }
+        // [END update_profile]
+    }
+
+    override fun updateEmail() {
+        // [START update_email]
+        val user = FirebaseAuth.getInstance().currentUser
+
+        user?.updateEmail("user@example.com")
+                ?.addOnCompleteListener { task ->
+                    if (task.isSuccessful) {
+                        Log.d(TAG, "User email address updated.")
+                    }
+                }
+        // [END update_email]
+    }
+
+    override fun updatePassword() {
+        // [START update_password]
+        val user = FirebaseAuth.getInstance().currentUser
+        val newPassword = "SOME-SECURE-PASSWORD"
+
+        user?.updatePassword(newPassword)
+                ?.addOnCompleteListener { task ->
+                    if (task.isSuccessful) {
+                        Log.d(TAG, "User password updated.")
+                    }
+                }
+        // [END update_password]
+    }
+
+    override fun sendEmailVerification() {
+        // [START send_email_verification]
+        val auth = FirebaseAuth.getInstance()
+        val user = auth.currentUser
+
+        user?.sendEmailVerification()
+                ?.addOnCompleteListener { task ->
+                    if (task.isSuccessful) {
+                        Log.d(TAG, "Email sent.")
+                    }
+                }
+        // [END send_email_verification]
+    }
+
+    override fun sendEmailVerificationWithContinueUrl() {
+        // [START send_email_verification_with_continue_url]
+        val auth = FirebaseAuth.getInstance()
+        val user = auth.currentUser
+
+        val url = "http://www.example.com/verify?uid=" + user!!.uid
+        val actionCodeSettings = ActionCodeSettings.newBuilder()
+                .setUrl(url)
+                .setIOSBundleId("com.example.ios")
+                // The default for this is populated with the current android package name.
+                .setAndroidPackageName("com.example.android", false, null)
+                .build()
+
+        user.sendEmailVerification(actionCodeSettings)
+                .addOnCompleteListener { task ->
```

Comment:
In this case you don't use the `?.` form for `user.sendEmailVerification` or `.addOnCompleteListener` but in all other cases you do.

We should be consistent, which is correct?

Reply:
@samtstern here the user was smart casted to be not null since I accidentally forgot to remove the double bang operator in line 140.

It should be user?. instead of user., you are correct.